### PR TITLE
Fixes #116 - Ldap integration test

### LINKED ISSUE #116: LDAP Authorization cache doesn't appear to be working
The changes implemented in https://github.com/lightblue-platform/lightblue-rest/pull/99 were recently deployed to our dev environment, and we lost all ability to access any of the REST endpoints (403 returned from JBoss in all cases).  Upon investigation, no information was being written to the access.log file, so it seemed like LDAP wasn't even being consulted for user roles.  I reverted the login module code to the version prior to the above referenced PR, and was able to access the services again.  So it appears that something related to authorization cache is not working correctly.  I didn't debug in detail, as I was primarily focused on restoring service in dev, so @luan-cestari, can you take a look at this when you have a chance?  Perhaps we can replicate the behavior in a unit test using an LDAP instance in order to get it resolved.

### PR DESCRIPTION
**Pull request by luan-cestari**
Related to this issue https://github.com/lightblue-platform/lightblue-rest/issues/116 (this PR also incorporate the changes of https://github.com/lightblue-platform/lightblue-rest/pull/117 )

### CODE REVIEW

**Comment by luan-cestari** [comment]
Fixed some log  dependency issues. Fixed the possible problem with cache that we found (but we didn't get evidences of the root cause). It need to finsih an unit test but I don't have enough knowledge about LDAP and Apache DS API, if someone could help with that, I can update this PR

**Comment by luan-cestari** [comment]
Some references that might be helpful: https://stackoverflow.com/questions/26398046/adding-a-new-attribute-to-an-object-class-and-expecting-it-to-automatically-show   http://jewzaam.gitbooks.io/lightblue-specifications/content/backend_specifications/ldap/crud_operation_examples.html   https://stackoverflow.com/questions/1560230/running-apache-ds-embedded-in-my-application   https://svn.apache.org/repos/asf/directory/apacheds/tags/1.5.3/core-integ/src/test/java/org/apache/directory/server/core/partition/PartitionIT.java    https://stackoverflow.com/questions/10353061/embedded-apacheds-in-my-application-it-fails-on-service-startup-defaultsch

**Comment by luan-cestari** [comment]
The unit test remaining just need to create a valid object into LDAP server to be queried

**Comment by jewzaam** [comment]
@luan-cestari does this fix #116?  And is there any more work to do before review? Your last comment makes me think something needs done with a unit test still.

**Comment by luan-cestari** [comment]
@jewzaam it probably solve it as we don't have enough data about the issue, so I wouldn't guarantee it solves the problem. The remaining test is something good to have, I ping some people on the channel for help, I think Derek is working on it.

File: auth/src/main/java/com/redhat/lightblue/rest/auth/ldap/LightblueLdapRoleProvider.java [hunk 1]
```diff
@@ -54,16 +53,21 @@ public LightblueLdapRoleProvider(String server, String searchBase, String bindDn
...
 
     @Override
     public List<String> getUserRoles(String userName) {
+        LOGGER.debug("Invoking LightblueLdapRoleProvider#getUserRoles");
         List<String> userRoles = new ArrayList<>();
-
         try {
-            userRoles.addAll(getUserRolesFromCache(userName));
+            List<String> userRolesFromCache = getUserRolesFromCache(userName);
+            if( userRolesFromCache != null && userRolesFromCache.isEmpty() ) {
```
Comment (jewzaam) [inline auth/src/main/java/com/redhat/lightblue/rest/auth/ldap/LightblueLdapRoleProvider.java]: Condition should be `!userRolesFromCache.isEmpty()` (missing !)

**Comment by jewzaam** [comment]
I'll merge after condition for using roles from cache is update.
Per note in this PR and @derek63 working on something for the unit test I am skipping review of the test.  I'll expect a separate PR coming with heavy updates.

**Comment by luan-cestari** [comment]
Fixed the line commented by @jewzaam

**Pull request merged**